### PR TITLE
Add support for transforming static string joins

### PR DESCRIPTION
**Pull request by akx**
This PR adds support for transforming static string join operations into f-strings.

IOW, 

```bash
$ flynt -tj -d test/test_static_join/victim.py
```

```diff
 a = "Hello"
-msg1 = " ".join([a, " World"])
-msg2 = "".join(["Finally, ", a, " World"])
-msg3 = "x".join(("1", "2", "3"))
-msg4 = "x".join({"4", "5", "yee"})
-msg5 = "y".join([1, 2, 3])  # Should be transformed
+msg1 = f"{a}  World"
+msg2 = f"Finally, {a} World"
+msg3 = "1x2x3"
+msg4 = "4x5xyee"
+msg5 = f"{1}y{2}y{3}"  # Should be transformed
 msg6 = a.join(["1", "2", "3"])  # Should not be transformed (not a static joiner)
 msg7 = "a".join(a)  # Should not be transformed (not a static joinee)
 msg8 = "a".join([a, a, *a])  # Should not be transformed (not a static length)
```

### CODE REVIEW

File: src/flynt/utils.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    conversion = -1 if conversion is None else ord(conversion.replace("!", ""))
+    return ast.FormattedValue(value=val, conversion=conversion, format_spec=format_spec)
+
+
+def ast_string_node(string: str) -> ast.Str:
+    return ast.Str(s=string)
+
+
+def fixup_transformed(tree: ast.AST) -> str:
```
Comment (ikamensh) [inline src/flynt/utils.py:42]: I don't understand the name of this function, why it's "fixup"? Can you please pick a more specific name and add a docstring

File: src/flynt/static_join/candidates.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        Finds all nodes that are joins with a static string literal
+        as the joiner and a static iterable as the joinee.
+        """
+        if get_static_join_bits(node) is not None:
+            self.victims.append(AstChunk(node))
+        else:
+            self.generic_visit(node)
+
+
+def join_candidates(code: str):
```
Comment (ikamensh) [inline src/flynt/static_join/candidates.py:27]: what is the advantage of making this a generator? can we return a collection?

File: test/test_static_join/victim.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+a = "Hello"
```
Comment (ikamensh) [inline test/test_static_join/victim.py:1]: So this is a demo file on which to test? Let's add a docstring on the top.

File: test/test_static_join/test_sj_transformer.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import pytest
+
+from flynt.static_join.transformer import transform_join
+
+pytestmark = pytest.mark.skipif(
+    sys.version_info < (3, 8), reason="requires python3.8 or higher"
+)
+
+CASES = [
```
Comment (ikamensh) [inline test/test_static_join/test_sj_transformer.py:12]: Previously a lot of edge cases shown up when using more diverse operations. Can you add:
* indexing (elements[123])
* function calls
* list / dict comprehensions
* lambda functions

I don't care so much if they are converted or not, but they should not crash :)

**Review by ikamensh — COMMENTED** [review]
Thanks and respect for understanding how to add a feature to this not-so-clean-architecture project. I've left some improvements requests, but this is on a good track.

File: src/flynt/api.py [hunk 5]
```diff
@@ -105,7 +115,13 @@ def default_result():
     return True, changes, len(contents), len(new_code)
 
 
-def fstringify_files(files, multiline, len_limit, transform_concat):
+def fstringify_files(
```
Comment (ikamensh) [inline src/flynt/api.py:118]: there are so-called integration tests that invoke file processing, can you add a test that without the `transform_join` flag there is no change? I think in current files there are just no candidate expressions, so they might not be able to tell how is software behaving in that regard..

File: src/flynt/utils.py (hunk 1 above)
Comment (akx) [reply src/flynt/utils.py:42]: This was extracted from https://github.com/ikamensh/flynt/blob/071dfb012adca6c13a311bc2f393b8a0907e9cc5/src/flynt/string_concat/transformer.py#L100-L106 because the same thing needs to be done with the static join transformer. "Fixup" seemed apt enough since it does some minor... well, fixups. 😅

File: src/flynt/static_join/candidates.py (hunk 2 above)
Comment (akx) [reply src/flynt/static_join/candidates.py:27]: We can, sure. I just basically copied this from https://github.com/ikamensh/flynt/blob/071dfb012adca6c13a311bc2f393b8a0907e9cc5/src/flynt/string_concat/candidates.py#L37-L45 :)

File: test/test_static_join/victim.py (hunk 3 above)
Comment (akx) [reply test/test_static_join/victim.py]: Resolved by getting rid of the whole file – it was just duplicating `CASES` anyway.

File: test/test_static_join/test_sj_transformer.py (hunk 4 above)
Comment (akx) [reply test/test_static_join/test_sj_transformer.py]: Done deal.

File: src/flynt/api.py (hunk 5 above)
Comment (akx) [reply src/flynt/api.py:118]: Sure, but maybe we should get #146 merged first to avoid double work in that regard?

Comment (akx) [reply src/flynt/api.py:118]: Rebased + added integration test that checks that a candidate join doesn't get changed by default.

File: src/flynt/utils.py (hunk 1 above)
Comment (ikamensh) [reply src/flynt/utils.py:42]: so maybe we can put in english what it's intention is: 1) remove trailing newline if present, 2) remove double escapes for newline that are artifact of python string representation and 3) replace triple quotes with double quotes (why? I don't remember). Not a blocker, can be added as a follow-up.

**Review by ikamensh — APPROVED** [review]
